futures-util/src/compat/tokio.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{future::FutureExt, try_future::TryFutureExt};
+use std::{
+    future::FutureObj,
+    task::{Executor, SpawnErrorKind, SpawnObjError},
+};
+use tokio_executor::{DefaultExecutor, Executor as _};
+
+/// An executor that delegates to `tokio`'s [`DefaultExecutor`], will panic if
+/// used in the context of a task that is not running on `tokio`'s executor.
+///
+/// # Examples
+///
+/// ```ignore
+/// #![feature(async_await, await_macro, underscore_imports, pin)]
+///
+/// use std::thread;
+/// use std::boxed::PinBox;
+///
+/// use futures::{
+///     channel::oneshot,
+///     future::FutureExt as _,
+///     future::TryFutureExt as _,
+///     executor::spawn,
+///     compat::TokioDefaultExecutor,
+/// };
+///
+/// let (sender, receiver) = oneshot::channel::<i32>();
+///
+/// let handle = thread::spawn(move || {
+///     futures::executor::block_on(async move {
+///         await!(receiver).unwrap()
```

Comment:
Why not `block_on(receiver).unwrap()` at the bottom? Does it have to be in another thread?